deploy/runtools/runtime_config.py

```diff
@@ -268,25 +295,34 @@ def __init__(self, runtimeconfigfile: str, configoverridedata: str) -> None:
             rootLogger.warning(overridefield + "=" + overridevalue)
             runtime_dict[overridesection][overridefield] = overridevalue
 
-        runfarmtagprefix = "" if 'FIRESIM_RUNFARM_PREFIX' not in os.environ else os.environ['FIRESIM_RUNFARM_PREFIX']
-        if runfarmtagprefix != "":
-            runfarmtagprefix += "-"
+        # Setup the run farm
+        defaults_file = runtime_dict['run_farm_config']['defaults']
+        with open(defaults_file, "r") as yaml_file:
+            run_farm_configfile = yaml.safe_load(yaml_file)
+        run_farm_type = run_farm_configfile["run_farm_type"]
+        run_farm_args = run_farm_configfile["args"]
 
-        self.runfarmtag = runfarmtagprefix + runtime_dict['run_farm']['run_farm_tag']
+        # add the overrides if it exists
 
-        aws_resource_names_dict = aws_resource_names()
-        if aws_resource_names_dict['runfarmprefix'] is not None:
-            # if specified, further prefix runfarmtag
-            self.runfarmtag = aws_resource_names_dict['runfarmprefix'] + "-" + self.runfarmtag
+        # taken from https://gist.github.com/angstwad/bf22d1822c38a92ec0a9
+        def deep_merge(a: dict, b: dict) -> dict:
+            result = deepcopy(a)
+            for bk, bv in b.items():
+                av = result.get(bk)
+                if isinstance(av, dict) and isinstance(bv, dict):
+                    result[bk] = deep_merge(av, bv)
+                else:
+                    result[bk] = deepcopy(bv)
+            return result
 
-        self.f1_16xlarges_requested = int(runtime_dict['run_farm']['f1_16xlarges']) if 'f1_16xlarges' in runtime_dict['run_farm'] else 0
-        self.f1_4xlarges_requested = int(runtime_dict['run_farm']['f1_4xlarges']) if 'f1_4xlarges' in runtime_dict['run_farm'] else 0
-        self.m4_16xlarges_requested = int(runtime_dict['run_farm']['m4_16xlarges']) if 'm4_16xlarges' in runtime_dict['run_farm'] else 0
-        self.f1_2xlarges_requested = int(runtime_dict['run_farm']['f1_2xlarges']) if 'f1_2xlarges' in runtime_dict['run_farm'] else 0
+        override_args = runtime_dict['run_farm_config'].get('override_args')
+        if override_args:
+            run_farm_args = deep_merge(run_farm_args, override_args)
```

Comment:
This is the "override" code, it only merges the dicts and not lists. Take the following case:

```
override_args:
    KEY0: VAL0
    KEY1:
      KEY2: VAL2
      KEY3: VAL3    
    KEY4:
      - val4
      - val5
 ```
 
 In this case, keys 0-3 could be merged with the default dict (aka. default args). But `KEY4` would completely override `KEY4` in the default dict. This would leave a list of `val4, val5` only. This matters for `run_farm_hosts` since that is also a list. Thus you need to re-specify all types/amts-to-spawn (in AWS case) if you override.

Reply:
Punting the description to the documentation PR.